src/components/ImageView/index.native.js

```diff
@@ -25,59 +25,30 @@ const propTypes = {
     /** Function for handle on press */
     onPress: PropTypes.func,
 
-    ...windowDimensionsPropTypes,
+    /** Additional styles to add to the component */
+    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
 };
 
 const defaultProps = {
     isAuthTokenRequired: false,
     onPress: () => {},
+    style: {},
 };
 
-class ImageView extends PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            isLoading: true,
-            imageWidth: 0,
-            imageHeight: 0,
-            interactionPromise: undefined,
-        };
-
-        // Use the default double click interval from the ImageZoom library
-        // https://github.com/ascoders/react-native-image-zoom/blob/master/src/image-zoom/image-zoom.type.ts#L79
-        this.doubleClickInterval = 175;
-        this.imageZoomScale = 1;
-        this.lastClickTime = 0;
-        this.amountOfTouches = 0;
-
-        // PanResponder used to capture how many touches are active on the attachment image
-        this.panResponder = PanResponder.create({
-            onStartShouldSetPanResponder: this.updatePanResponderTouches.bind(this),
-        });
-
-        this.configureImageZoom = this.configureImageZoom.bind(this);
-        this.imageLoadingStart = this.imageLoadingStart.bind(this);
-    }
-
-    componentDidUpdate(prevProps) {
-        if (this.props.url === prevProps.url) {
-            return;
-        }
+const DOUBLE_CLICK_INTERVAL = 175;
 
-        this.imageLoadingStart();
+function ImageView({isAuthTokenRequired, url, onScaleChanged, onPress, style}) {
+    const {windowWidth, windowHeight} = useWindowDimensions();
 
-        if (this.interactionPromise) {
-            this.state.interactionPromise.cancel();
-        }
-    }
+    const [isLoading, setIsLoading] = React.useState(true);
+    const [imageWidth, setImageWidth] = React.useState(0);
+    const [imageHeight, setImageHeight] = React.useState(0);
+    const [containerHeight, setContainerHeight] = React.useState();
 
-    componentWillUnmount() {
-        if (!this.state.interactionPromise) {
-            return;
-        }
-        this.state.interactionPromise.cancel();
-    }
+    const imageZoomScale = React.useRef(1);
+    const lastClickTime = React.useRef(0);
+    const numberOfTouches = React.useRef(0);
+    const zoom = React.useRef();
```

Comment:
Should it have any default value. I prefer to avoid undefined.
```suggestion
    const zoom = React.useRef(0/null/other value);
```